### ISSUE TITLE
None is not considered Hashable

### ISSUE THREAD

**Issue by NeilGirdhar** [python/mypy#8768]
```python
from typing import Hashable


def f() -> Hashable:
    return None


print(isinstance(None, Hashable))
```
The print returns True, so Python understands that None is hashable, but MyPy complains about incompatible return types:
```
error: Incompatible return value type (got "None", expected "Hashable")
```

**Comment by JukkaL** [comment]
Marking as high priority since typeshed is getting new `Hashable` type annotations, and these can cause false positives when updating to a more recent mypy release.
